TL321x Platform Bringup

liucoldstar opened the pull request:
For compliance check only.

damien0x0023 committed:
soc: riscv: telink: telink_tlx: Create SoC Zephyr Skeleton for TL321x

    1. Add new SoC Telink TLx(telink_tlx) based on Telink B9x(telink_b9x).
    2. Add new board tl3218x based on tlsr9528a.
    3. Add Zephyr driver and DTS files for TLx SoC.
    4. Add tl3218x.& tl3218x.overlay in ot-cli samples to apply uart.
    5. Update hal_telink to match tl321x driver skeleton.
    6. Update openthread_telink_lib to match new SoC series.

    Known issues:

    1. USB driver does not work normally.
       usb-cdm feature needs to be applied.
    2. Remove Deep Retention samples.
       They may not be able to build due to missing definitions.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

damien0x0023 committed:
telink: hal: update west.yml

1. Fix missing condition for retention apps.
2. Add retention apps back.
3. Remove unnecessary conditions.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

liucoldstar committed:
telink: tl321x: Initial working code for blinky

- Perform minimal code changes to make the blinky sample functional.
- Update the link to hal_telink.

Signed-off-by: Andy Liu <liucoldstar@gmail.com>

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/36283106d9b0b1d5fb3025a98be5a58007ac3d77 (develop,telink_tl321x) | https://github.com/telink-semi/hal_telink/commit/1da8d34e94c06f4a5161c9d9e615cf3c2da5feae (tl321x_bringup) | [telink-semi/hal_telink@36283106..1da8d34e](https://github.com/telink-semi/hal_telink/compare/36283106d9b0b1d5fb3025a98be5a58007ac3d77..1da8d34e94c06f4a5161c9d9e615cf3c2da5feae) |
| openthread_telink_lib | https://github.com/telink-semi/openthread_telink_lib/commit/80645ba0b4b589afd8c0712cb1a01091f4b35ab8 (main) | https://github.com/telink-semi/openthread_telink_lib/commit/84e12fd24613b2871c47378c39c2f44447f528ba (tl321x_skeleton) | [telink-semi/openthread_telink_lib@80645ba0..84e12fd2](https://github.com/telink-semi/openthread_telink_lib/compare/80645ba0b4b589afd8c0712cb1a01091f4b35ab8..84e12fd24613b2871c47378c39c2f44447f528ba) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

andy-liu-telink committed:
telink: tl321x: update workflow and fix compliance issues

- update the workflow for tl321x
- fix compliance issues

Signed-off-by: Andy Liu <jie.liu@telink-semi.com>

andy-liu-telink committed:
telink: tl321x: update the hal_telink revision

- update the hal_telink revision

Signed-off-by: Andy Liu <jie.liu@telink-semi.com>

damien0x0023 committed:
telink: tl321x: update led, button, gpio definitions

  - update gpio drivers.
  - update dts/dtsi files.
  - revert blinky sample to ensure it work.
  - update pinctrl files.
  - set year to 2024 for tl321x-related files.
  - fix code style to pass CI checks.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

andy-liu-telink reviewed (approved):
Approve
